stdlib/_interpchannels.pyi

```diff
@@ -0,0 +1,86 @@
+from _typeshed import structseq
+from typing import Any, Final, Literal, TypeVar, final
+from typing_extensions import Buffer, Self, TypeAlias
+
+_T = TypeVar("_T")
+_SendType: TypeAlias = Literal["send", "recv", "both"]
+
+class ChannelError(RuntimeError): ...
+class ChannelClosedError(ChannelError): ...
+class ChannelEmptyError(ChannelError): ...
+class ChannelNotEmptyError(ChannelError): ...
+class ChannelNotFoundError(ChannelError): ...
+
+@final
+class ChannelID:
+    @property
+    def end(self) -> _SendType: ...
+    @property
+    def send(self) -> Self: ...
+    @property
+    def recv(self) -> Self: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ge__(self, other: object) -> bool: ...
+    def __gt__(self, other: object) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __index__(self) -> int: ...
+    def __int__(self) -> int: ...
+    def __le__(self, other: object) -> bool: ...
+    def __lt__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+
+@final
+class ChannelInfo(structseq[Any], tuple[bool, bool, bool, int, int, int, int, int]):
+    __match_args__: Final = (
+        "open",
+        "closing",
+        "closed",
+        "count",
+        "num_interp_send",
+        "num_interp_send_released",
+        "num_interp_recv",
+        "num_interp_recv_released",
+    )
+    @property
+    def open(self) -> bool: ...
+    @property
+    def closing(self) -> bool: ...
+    @property
+    def closed(self) -> bool: ...
+    @property
+    def count(self) -> int: ...  # type: ignore[override]
+    @property
+    def num_interp_send(self) -> int: ...
+    @property
+    def num_interp_send_released(self) -> int: ...
+    @property
+    def num_interp_recv(self) -> int: ...
+    @property
+    def num_interp_recv_released(self) -> int: ...
+    @property
+    def num_interp_both(self) -> int: ...
+    @property
+    def num_interp_both_recv_released(self) -> int: ...
+    @property
+    def num_interp_both_send_released(self) -> int: ...
+    @property
+    def num_interp_both_released(self) -> int: ...
+    @property
+    def recv_associated(self) -> bool: ...
+    @property
+    def recv_released(self) -> bool: ...
+    @property
+    def send_associated(self) -> bool: ...
+    @property
+    def send_released(self) -> bool: ...
+
+def create() -> ChannelID: ...
+def destroy(cid: int) -> None: ...
+def list_all() -> list[int]: ...
```

Comment:
This doesn't look correct to me:

```pycon
>>> import _interpchannels
>>> _interpchannels.create()
ChannelID(0)
>>> _interpchannels.list_all()
[ChannelID(0)]
>>> type(_[0])
<class '_interpchannels.ChannelID'>
```

```suggestion
def list_all() -> list[ChannelID]: ...
```

Reply:
Let me take another look, perhaps because I wasn't on the latest build (I was using Alpha 3), or just missed that.

Reply:
I think I just missed that for some reason. Fixed: https://github.com/python/typeshed/pull/12199/commits/cf8580e6c61a1f5877605cb6bf1a685ef72bc9e2